sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala

```diff
@@ -1,112 +1,178 @@
+    val blockId = nodeView.history.getBlockIdByHeight(Integer.parseInt(Numeric.cleanHexPrefix(tag.getValue), 16))
+    val txHashes: ListBuffer[String] = ListBuffer()
+    val txViews: ListBuffer[EthereumTransactionView] = ListBuffer()
+
+    if (blockId.isEmpty) return new EthereumBlock()
+
+    val txs = getTransactionsFromBlock(getBlockById(blockId.get()))
+    txs.foreach(transaction => {
+      txViews.append(new EthereumTransactionView(transaction))
+      txHashes.append(Numeric.prependHexPrefix(transaction.id))
+    })
+    new EthereumBlock(tag.getValue, Numeric.prependHexPrefix(blockId.get()), txHashes.toList.asJava, txViews.toList.asJava, hydratedTx, nodeView.history.getBlockById(blockId.get()).get())
+  }
+
+  def getTransactionsFromBlock(block: AccountBlock) = {
+    val ethTx = new ListBuffer[EthereumTransaction]
+    block.transactions.foreach(transaction =>
+      if (transaction.isInstanceOf[EthereumTransaction]) ethTx.append(transaction.asInstanceOf[EthereumTransaction]))
+    ethTx
+  }
+
+  def getBlockById(blockId: String) = nodeView.history.getBlockById(blockId).get()
+
+  @RpcMethod("eth_getBlockByHash") def getBlockByHash(tag: Quantity, hydratedTx: Boolean): EthereumBlock = {
+    val txHashes: ListBuffer[String] = ListBuffer()
+    val txViews: ListBuffer[EthereumTransactionView] = ListBuffer()
+
+    val txs = getTransactionsFromBlock(getBlockById(Numeric.cleanHexPrefix(tag.getValue)))
+    txs.foreach(transaction => {
+      txViews.append(new EthereumTransactionView(transaction))
+      txHashes.append(Numeric.prependHexPrefix(transaction.id))
+    })
+    new EthereumBlock(Numeric.prependHexPrefix(Integer.toHexString(nodeView.history.getBlockHeight(Numeric.cleanHexPrefix(tag.getValue)).get())), Numeric.prependHexPrefix(tag.getValue), txHashes.toList.asJava, txViews.toList.asJava, hydratedTx, nodeView.history.getBlockById(Numeric.cleanHexPrefix(tag.getValue)).get())
+  }
+
+  @RpcMethod("eth_call") def call(params: TransactionArgs, tag: Quantity): Data = {
+    val stateDbRoot = getStateDbRootFromBlockById(getBlockIdByTag(tag.getValue))
+    if (stateDbRoot == null) return new Data(null)
+    val result = Evm.Apply(
+      stateDbRoot.getStateDbHandle,
+      params.getFrom,
+      if (params.to == null) null else params.to.toBytes,
+      params.value,
+      params.getData,
+      params.gas,
+      params.gasPrice
+    )
+
+    if (result.evmError.nonEmpty) {
+      throw new RpcException(
+        new RpcError(
+          RpcCode.ExecutionError.getCode,
+          result.evmError,
+          Numeric.toHexString(result.returnData)
+        )
+      )
     }
-    return new Data(null)
+    new Data(result.returnData)
   }
 
   @RpcMethod("eth_blockNumber") def blockNumber = new Quantity(Numeric.toHexStringWithPrefix(BigInteger.valueOf(getBlockHeight)))
 
+  private def getBlockHeight = nodeView.history.getCurrentHeight
```

Comment:
this method is used only in the `def blockNumber`.
let's remove it at all. the name `getBlockHeight` confuse me a bit.